### ISSUE TITLE
 Documentation of cell neighbors

### ISSUE THREAD

**Issue by rgoermer** [dealii/dealii#17073]
Dear folks, 

The documentation for [`CellAccessor<dim, spacedim>::neighbor(const unsigned int)`](https://www.dealii.org/current/doxygen/deal.II/classCellAccessor.html#a177d9a97d1a0d8a066e45040c0b6aac9) is whacky in my opinion. This line

> The neighbor of a cell has at most the same level as this cell.

stands somewhat in contradiction to what is stated in the example

> In other words, the cell you get as neighbor has the same refinement level as the one you're on right now

This misunderstanding could be lifted if a different example with anisotropic refinements is considered. Let 
![grid-3-ref-2](https://github.com/dealii/dealii/assets/77726182/9b58d18c-8e40-4f6c-b612-890c7c0e8733)
be the refinement of 
![grid-3-ref-1](https://github.com/dealii/dealii/assets/77726182/d0845360-0d4a-4bb9-b5d5-facddc2876e8)

Also, including level and index description of the cells makes it easier to describe what is going on. Denote by `cell[level , ind]` a cell with level `level` and index `ind`. Then, `neighbor(cell[1, 3], 0) = cell[2, 7]` is returned. With the above statements, I had assumed that the parent of `cell[2, 7]` would be returned as `cell[2, 7]` has level 2 and `cell[1, 3]` has level 1.   

In conclusion, I think it is best to state that `cell -> neighbor(face_no)` returns the finest cell that wholly contains `cell -> face(face_no)` and that is no superset of `cell`. Mathematically I guess you may want to write something along the lines of 
$$\mathcal{N}_f(Q) = Q' \text{ with } Q\not\subset Q' \text{ and } Q|_f\subset \partial Q$$

I apologize, If this is the wrong page to notify you, in which case I will gladly switch to the Google mailing list.

Kind regards
Robin
